src/spoke/VaultRegistry.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IShareToken} from "./interfaces/IShareToken.sol";
+import {IVault, VaultKind} from "./interfaces/IVault.sol";
+import {VaultDetails, ISpoke} from "./interfaces/ISpoke.sol";
+import {IVaultRegistry} from "./interfaces/IVaultRegistry.sol";
+import {IVaultFactory} from "./factories/interfaces/IVaultFactory.sol";
+import {IVaultManager, REQUEST_MANAGER_V3_0} from "./interfaces/legacy/IVaultManager.sol";
+
+import {Auth} from "../misc/Auth.sol";
+import {Recoverable} from "../misc/Recoverable.sol";
+
+import {PoolId} from "../common/types/PoolId.sol";
+import {AssetId} from "../common/types/AssetId.sol";
+import {ShareClassId} from "../common/types/ShareClassId.sol";
+import {VaultUpdateKind} from "../common/libraries/MessageLib.sol";
+import {IRequestManager} from "../common/interfaces/IRequestManager.sol";
+import {IVaultRegistryGatewayHandler} from "../common/interfaces/IVaultRegistryGatewayHandler.sol";
+
+/// @title  VaultRegistry
+/// @notice This contract manages vault deployment, linking, and unlinking operations for pools and share classes
+contract VaultRegistry is Auth, Recoverable, IVaultRegistry, IVaultRegistryGatewayHandler {
+    mapping(
+        PoolId poolId => mapping(ShareClassId scId => mapping(AssetId assetId => mapping(IRequestManager => IVault)))
+    ) public vault;
+
+    mapping(IVault => VaultDetails) internal _vaultDetails;
+
+    ISpoke public spoke;
```

Comment:
Nit: Shouldn't this be at top of the storage ordering by our convention? And maybe also `_vaultDetails` in the middle.